### DISCUSSION TITLE
Multicast Delegate shorter syntax

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/csharplang#1646] [Language Ideas]
I have this delegate:
```public delegate void MsgSub(string Msg);```

suppose I have two voids Show1 and show2, then I can do:
```
MsgSub M = new MsgSub(Show1) + new MsgSub(Show2) + new MsgSub(Show3);
```

I want to do this directly:
```
MsgSub M = Show1 + Show2 + Show3;
```

Currently this gives a syntax error, so it will not break anything.

**Comment by svick** [comment]
[You can do:](https://sharplab.io/#v2:C4LgbgpgTglgZgTwD4AEBMACAwgWAFADe+GJGKALBgLIAUAlMaUXqa9QM4DmAygK4BG1DAF4MAOwgB3DjwE1uACwD2kgIx0MAagyKVmbbskBmANyMSAX3MZrFHcrU0UqgAwyNBKy1J3DaJ67uGJ62lIZGAW5UXB5eXvgoRhgAJhAANhCcAIbAEGSU0bL8ke4mQA=)

```c#
MsgSub M = new MsgSub(Show1) + Show2 + Show3;
```

Is that not good enough?

In any case, I'm not convinced that multicast delegates are a feature that needs more syntax sugar, since I think it's not commonly used on its own.

**Comment by ghost** [comment]
@svick 
> Is that not good enough?

I'd normally say no, but I faced such syntax in Verex:
```
var pattern = Text("a") + "xyx";
```

I thought C# gave especial syntax to combine delegates, since It doesn't work in VB.NET!. Now I wonder why?!

**Comment by AustinBryan** [comment]
Also, with [#100](https://github.com/dotnet/csharplang/issues/100), the code can become:

```csharp
MsgSub M = new(Show1) + Show2 + Show3;
```

And, you could also create some global function: 

```csharp
MsgSub M = GlobalFunctions.MsgSub(Show1, Show2, Show3);
```

Or:


```csharp
using static GlobalFunctions;

...

MsgSub M = MsgSub(Show1, Show2, Show3);
```

Which is a bit longer than the first example, but easier to type.
